addons/mail/static/src/new/core/chat_window_model.js

```diff
@@ -32,8 +35,25 @@ export class ChatWindow {
             _state: state,
         });
         this.update(data);
-        state.chatWindows.push(this);
-        return state.chatWindows.find((c) => c.threadLocalId === data.threadLocalId); // return reactive version
+        let idx;
+        if (!data.replaceNewMessageChatWindow) {
+            idx = state.chatWindows.length;
+        } else {
+            const newMessageChatWindowIndex = state.chatWindows.findIndex(
+                ({ threadLocalId }) => !threadLocalId
+            );
```

Comment:
Could just check on `thread`, it's less intimidating to read rather than `threadLocalId`